src/TailwindBinary.php

```diff
@@ -95,7 +95,29 @@ private function downloadExecutable(): void
 
     private function getVersion(): string
     {
-        return $this->cachedVersion ??= $this->binaryVersion ?? $this->getLatestVersion();
+        return $this->cachedVersion ??= $this->binaryVersion
+            ?? $this->getLatestInstalledVersion()
+            ?? $this->getLatestVersion();
+    }
+
+    private function getLatestInstalledVersion(): ?string
+    {
+        $binaryName = self::getBinaryName();
+
+        $installedBinaries = glob("{$this->binaryDownloadDir}/v*/{$binaryName}");
+
+        $installedVersions = array_map(function (string $path) {
+            return basename(\dirname($path));
+        }, $installedBinaries);
+        sort($installedVersions);
```

Comment:
I suppose it may not sort correct in some cases well:
<img width="597" alt="Screenshot 2025-01-29 at 13 18 51" src="https://github.com/user-attachments/assets/38b91c70-21ae-4fda-af82-a172f4c0d106" />

https://3v4l.org/8CS3R#v8.4.3

Probably sorting is pointless in this case?

Reply:
You were correct that it wasn't sorting properly before, however I would point out that I am sorting the directories the binaries are in, not the binaries themselves, which should be just the version number.

```bash
$ tree ./var/tailwind/
./var/tailwind/
├── app.built.css
├── v3.4.17
│   └── tailwindcss-linux-x64
└── v4.0.0
    └── tailwindcss-linux-x64
```

changing `sort($installedVersions)` to `usort($installedVersions, 'version_compare')` should do the trick.

Reply:
Tested it in my symfony project, and it worked correctly :)